src/nupic/algorithms/Connections.cpp

```diff
@@ -85,7 +85,9 @@ Synapse Connections::createSynapse(const Segment& segment,
   vector<SynapseData>& synapses = cells_[segment.cell.idx].segments[segment.idx].synapses;
   if (synapses.size() == maxSynapsesPerSegment_)
   {
-    NTA_THROW << "Cannot create synapse: segment has reached maximum number of synapses.";
+    Synapse minSynapse;
+    minPermanenceSynapse(segment, minSynapse);
```

Comment:
I considered this. However, it will only return false if `synapses.size()` == 0, which can only be true at this point in the program flow if `maxSynapsesPerSegment_ == 0`, which is an invalid parameter value, so I decided that the check wasn't necessary.

I can see removing the turn value from `minPermanenceSynapse` though.